backend/src/tests/tests.test.js

```diff
@@ -3,27 +3,85 @@ const { ApolloServer, gql } = require("apollo-server");
 const {beforeEach, describe, expect, it} = require("@jest/globals");
 const PostsAPI = require("../datasources/postApi");
 const UsersAPI = require("../datasources/userApi");
+const AuthAPI = require("../datasources/authenticationApi");
 const typeDefs = require('../typeDefs');
 const resolvers = require('../resolvers');
+const permissions = require('../security/permissions');
+const { v4: uuidv4 } = require('uuid');
+const { applyMiddleware } = require('graphql-middleware');
+const { makeExecutableSchema } = require('graphql-tools');
+const jwt = require('jsonwebtoken');
 
 let usersApi = new UsersAPI();
 let postsApi = new PostsAPI();
+let authApi = new AuthAPI();
+
+const schema = applyMiddleware(
+    makeExecutableSchema({
+        typeDefs,
+        resolvers,
+    }),
+    permissions,
+);
+
+let reqMock;
+let resMock;
+const context = require('../context');
+const contextPipeline = () => context({ req: reqMock, res: resMock });
 
 const server = new ApolloServer({
-    typeDefs, 
-    resolvers, 
+    schema,
+    context: contextPipeline,
     dataSources: () => {
         return {
           usersApi: usersApi,
-          postsApi: postsApi
+          postsApi: postsApi,
+          authApi: authApi
         }
     }
 });
 
+let mike_auth_token = '';
+let atanas_auth_token = '';
+
 const { query, mutate } = createTestClient(server);
 
-describe("query", () => {
+beforeEach(async () => {
+    process.env.JWT_SECRET='TEST_SECRET';
+    // For me a mock express.Request containing some headers is enough.
+    // I assume typescript will complain here:
+    reqMock = { headers: {} };
+    resMock = {};
+
+    let userIds = [uuidv4(), uuidv4()];
+
+    mike_auth_token = await authApi.createToken(userIds[0]);
+    atanas_auth_token = await authApi.createToken(userIds[1]);
 
+    postsApi.posts = [
+        { title: "Mike's Post 1", votes: 0, upvoters:[], author: userIds[0] },
+        { title: "Mike's Post 2", votes: 0, upvoters:[], author: userIds[0] },
+        { title: "Atanas's Post 1", votes: 0, upvoters:[], author: userIds[1] },
+        { title: "Atanas's Post 2", votes: 0, upvoters:[], author: userIds[1] },
```

Comment:
```suggestion
        new Post({ title: "Mike's Post 1", author: userIds[0] }),
        new Post({ title: "Mike's Post 2", author: userIds[0] }),
        new Post({ title: "Atanas's Post 1", author: userIds[1] }),
        new Post({ title: "Atanas's Post 2", author: userIds[1] }),
```
This could be DRYed with a class.